benchmarks/benchmarks/pybench/persistence/common.py

```diff
@@ -0,0 +1,65 @@
+import abc
+import datetime
+from collections.abc import Iterable, Sequence
+from dataclasses import dataclass
+from functools import cached_property
+from pathlib import Path
+from typing import Any, Optional, Protocol, TypedDict
+
+
+@dataclass(frozen=True)
+class BenchMeta:
+    benchmark_name: str
+    benchmark_subname: str
+
+
+class BenchSchemaProto(Protocol):
+    tags: Iterable[str]
+    base: str
+
+
+class BenchAccessProto(Protocol):
+    meta: BenchMeta
+    data_dir: Path
+
+    @abc.abstractmethod
+    @cached_property
+    def schemas(self) -> Sequence[BenchSchemaProto]:
+        ...
+
+    @abc.abstractmethod
+    def get_id(self, schema) -> str:
+        ...
+
+
+class BenchRecord(TypedDict):
+    is_actual: bool
+    benchmark_name: str
+    benchmark_subname: str
+    base: str
+    local_id: str
+    global_id: str
+    tags: str
+    kwargs: str
+    distributions: str
+    data: str
+    created_at: datetime.datetime
```

Comment:
I think, it would be better to pass `kwargs`, `distributions` and `data` as dicts and lists (serialization is responsibility of operator).

Also, all these types should be stores as JSON insize SQLite